src/engine/Engine.cpp

```diff
@@ -1951,10 +1951,9 @@ void Engine::updateDirections()
                 constraint->updateDirection();
 }
 
-void Engine::updateScores()
+void Engine::updateScores( DivideStrategy strategy )
 {
-    if ( _networkLevelReasoner &&
-         GlobalConfiguration::SPLITTING_HEURISTICS == DivideStrategy::Polarity )
+    if ( _networkLevelReasoner && strategy == DivideStrategy::Polarity )
     {
         // We find the earliest K ReLUs that have not been fixed, update
```

Comment:
Hmm... Could the if and else branches be reversed? I'd expect EarliestRelu to use the topological order, not Polarity

Reply:
Both the Polarity and the EarliestReLU use the topological order. I refactored the code a bit to make it cleaner.